src/messages/MMgrConfigure.h

```diff
@@ -36,7 +36,7 @@ class MMgrConfigure : public MessageInstance<MMgrConfigure> {
   // Default 0 means if unspecified will include all stats
   uint32_t stats_threshold = 0;
 
-  std::list<OSDPerfMetricQuery> osd_perf_metric_queries;
+  std::map<OSDPerfMetricQuery, std::list<OSDPerfMetricLimit>> osd_perf_metric_queries;
```

Comment:
Design question: why separate the limits from the query? Why not include the limits within the query?

Reply:
This looks cleaner to me: queries describe how to collect data, and limits how to limit the result sent by osds. In `OSDPerfMetricCollector`, we really need limits to be separated from the queries because we need them to be indexed by `query_id`. On the OSD side it is not so clear, but still you don't need to pass limits to every pg. You need them only when the OSD is building a report from the collected stats.

I understand that having limts as a part of `OSDPerfMetricQuery` would be nicer in this particular (`MMgrConfigure`) case. But it looks to me it would make things a little weird in other cases: should the limits filed be empty for the queries stored in  `OSDPerfMetricCollector`, because we would need to store them separately anyway? In the query comparison operators show we compare the limits?

BTW, I just realized that the list of limits in osd_perf_metric_queries may contain duplicates, which I need to filter out (or use `std::set` instead of `std::list`).

Reply:
updated, replacing `std::list<OSDPerfMetricLimit>` with `std:set<OSDPerfMetricLimit>`.